### PR TITLE
Default pod restart policy to OnFailure

### LINKED ISSUE #61: Default restartPolicy
Job only accepts only Never or OnFailure restartPolicy. We can default it to Never since most HPC and ML training workloads don't tolerate failures and can only recover by restarting the whole job

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Fixes #61 

I also bumped the e2e test timeout to avoid flakiness.

### CODE REVIEW

File: api/v1alpha1/jobset_webhook_test.go [hunk 1]
```diff
@@ -21,7 +33,11 @@ func TestJobSetDefaulting(t *testing.T) {
 					ReplicatedJobs: []ReplicatedJob{
 						{
 							Template: batchv1.JobTemplateSpec{
```
Comment (kannon92) [inline api/v1alpha1/jobset_webhook_test.go:35]: Could you create this template once and reuse everywhere?

Comment (danielvegamyhre) [reply api/v1alpha1/jobset_webhook_test.go:37]: makes sense, done

**Pull request merged**